documentation/documentation_compiler.ts

```diff
@@ -32,32 +33,62 @@ function compileSnippets() {
 function compileExamples() {
   const compiledExamples: CompiledExample[] = Examples.map(example => {
     const content = getContentFile(example.contentFile);
-    const code = compileCodeFiles(example);
+    const compiledExampleSnippets = compileExampleSnippets(example);
+    if (!isValidReferencePath(example.referencePath)) {
+      throw new Error(`${example.referencePath} is not a valid path`);
+    }
     return {
+      tokens: example.tokens,
+      referencePath: example.referencePath,
       content,
-      code,
-      categories: example.categories,
-      triggerTokens: example.triggerTokens,
+      exampleSnippets: compiledExampleSnippets,
     };
   });
+
   fs.writeFileSync(path.join(DocumentationRoot, 'generated/examples.json'), JSON.stringify(compiledExamples, null, 2));
 }
 
+function getCodeFile(file: string) {
+  const data = fs.readFileSync(path.join(DocumentationRoot, file), 'utf8');
+  const codeStart = data.indexOf(CodeBegin) + CodeBegin.length;
+  return data.substring(codeStart).trim();
+}
+
 function getContentFile(file: string) {
   return fs.readFileSync(path.join(DocumentationRoot, file), 'utf8').trim();
 }
 
-function compileCodeFiles(example: Example) {
-  const code = example.codeFiles.map(codeFile => {
-    return getCodeFile(codeFile);
+function compileExampleSnippets(example: Example): CompiledExampleSnippet[] {
+  return example.exampleSnippets.map(exampleSnippet => {
+    return {
+      name: formatSnippetName(path.basename(exampleSnippet.codeFile)),
```

Comment:
I would think the name should just be hand-written in the snippet definition.

Reply:
I thought it made sense for the example snippet file name to also be the name of the example.

Reply:
Yeah, they will likely track pretty closely in most cases, but it's a bit unintuitive that your filename will become a prettified name, and it eliminates the ability to do basic things like put non-alphanumeric characters in the title. Say I wanted a title that said "Dynamic Sync Tables (Advanced)", that would be impossible. Generally I prefer that easy things are easy to accomplish, so I'm hesitant for a system where somebody asks for a parenthesis in a title and we have to tell them they're out of luck.

Reply:
Gotcha, def more flexible to have a separate name property. I'll make that change